### PR TITLE
The null ArtifactsPath value causes exception for all EtwProfiler

### PR DESCRIPTION
**Pull request by wojtpl2**
Some improvements for #1235. 
This is just nicer error handling. Unfortunately, this will not correct the mistake itself.

### CODE REVIEW

**Comment by wojtpl2** [comment]
The last commit resolve problem from #1235.
@adamsitnik It is ready for review and merge 😉

File: src/BenchmarkDotNet/Configs/ImmutableConfigBuilder.cs [hunk 1]
```diff
@@ -61,7 +61,7 @@ public static ImmutableConfig Create(IConfig source)
                 uniqueRules,
                 uniqueRunnableJobs,
                 source.UnionRule,
-                source.ArtifactsPath,
+                source.ArtifactsPath ?? DefaultConfig.Instance.ArtifactsPath,
```
Comment (adamsitnik) [inline src/BenchmarkDotNet/Configs/ImmutableConfigBuilder.cs]: @wojtpl2 could you please add a test to `ImmutableConfigTests` that ensures that `ImmutableConfig.ArtifactsPath` is never null? I just want to make sure we don't regress in the future.

**Comment by wojtpl2** [comment]
@adamsitnik I've added tests for empty ArtifactsPath and Orderer. 

Sorry that I've removed trailing whitespaces. If you want, I can undo these changes.
I have https://marketplace.visualstudio.com/items?itemName=MadsKristensen.TrailingWhitespaceVisualizer

**Comment by adamsitnik** [comment]
> Sorry that I've removed trailing whitespaces

No worries, GH allows me to hide the whitespace changes:

obraz

**Pull request merged**